src/audio/module_adapter/module_adapter.c

```diff
@@ -1072,7 +1037,8 @@ int module_adapter_trigger(struct comp_dev *dev, int cmd)
 	}
 
 #if CONFIG_IPC_MAJOR_3
-	if (mod->num_input_buffers > 1) {
+	/* check if we do have more than 1 input buffer */
+	if (!list_item_is_last(&dev->bsource_list, &dev->bsource_list)) {
```

Comment:
is this change iso-functionality? This doesn't follow the replacement pattern in this patch.